### PR TITLE
V12 + Nginx

### PR DESCRIPTION
**Pull request by tonybenoy**
@Yenthe666

### CODE REVIEW

File: odoo_install.sh [hunk 1]
```diff
@@ -248,6 +256,88 @@ sudo update-rc.d $OE_CONFIG defaults
 
 echo -e "* Starting Odoo Service"
 sudo su root -c "/etc/init.d/$OE_CONFIG start"
+#--------------------------------------------------
+# Install Nginx if needed
+#--------------------------------------------------
+if [ $INSTALL_NGINX = "True" ]; then
+  echo -e "\n---- Installing and setting up Nginx ----"
+  sudo apt install nginx -y
+  cat <<EOF > ~/odoo
```
Comment (Yenthe666) [inline odoo_install.sh]: Why not use our variable `OE_USER` here?

File: odoo_install.sh [hunk 2]
```diff
@@ -248,6 +256,88 @@ sudo update-rc.d $OE_CONFIG defaults
...
+   # Add Headers for odoo proxy mode
+   proxy_set_header X-Forwarded-Host \$host;
+   proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
+   proxy_set_header X-Forwarded-Proto \$scheme;
+   proxy_set_header X-Real-IP \$remote_addr;
+   add_header X-Frame-Options "SAMEORIGIN";
+   add_header X-XSS-Protection "1; mode=block";
+
+   #   odoo    log files
+   access_log  /var/log/nginx/odoo-access.log;
```
Comment (Yenthe666) [inline odoo_install.sh]: Why not use our variable `OE_USER` here?

File: odoo_install.sh [hunk 3]
```diff
@@ -248,6 +256,88 @@ sudo update-rc.d $OE_CONFIG defaults
...
+   error_log       /var/log/nginx/odoo-error.log;
```
Comment (Yenthe666) [inline odoo_install.sh]: Why not use our variable `OE_USER` here?

File: odoo_install.sh [hunk 4]
```diff
@@ -248,6 +256,88 @@ sudo update-rc.d $OE_CONFIG defaults
...
+   proxy_cache_valid 200 302 60m;
+   proxy_cache_valid 404      1m;
+   proxy_buffering    on;
+   expires 864000;
+   proxy_pass    http://127.0.0.1:$OE_PORT;
+   }
+   }
+EOF
+
+  sudo mv ~/odoo /etc/nginx/sites-available/
```
Comment (Yenthe666) [inline odoo_install.sh]: Not sure but do we really need `sudo` for all these operations?

File: odoo_install.sh [hunk 5]
```diff
@@ -258,4 +348,7 @@ echo "Addons folder: $OE_USER/$OE_CONFIG/addons/"
 echo "Start Odoo service: sudo service $OE_CONFIG start"
 echo "Stop Odoo service: sudo service $OE_CONFIG stop"
 echo "Restart Odoo service: sudo service $OE_CONFIG restart"
+if [ $INSTALL_NGINX = "True" ]; then
+ echo "Done! The Nginx server is up and running. Configuration can be found at /etc/nginx/sites-available/odoo"
```
Comment (Yenthe666) [inline odoo_install.sh]: Why not use our variable `OE_USER` here?

File: odoo_install.sh [hunk 6]
```diff
@@ -61,7 +64,7 @@ sudo su - postgres -c "createuser -s $OE_USER" 2> /dev/null || true
 # Install Dependencies
 #--------------------------------------------------
 echo -e "\n--- Installing Python 3 + pip3 --"
-sudo apt-get install python3 python3-pip
+sudo apt-get install python3 python3-pip -y
```
Comment (Yenthe666) [inline odoo_install.sh:67]: 👍 good.

**Review by Yenthe666 — CHANGES_REQUESTED** [review]

**Comment by tonybenoy** [comment]
Thanks for the changes @Yenthe666 . I'll  test them out and get back to you :+1:

File: odoo_install.sh [hunk 7]
```diff
@@ -328,10 +328,10 @@ if [ $INSTALL_NGINX = "True" ]; then
    }
 EOF
 
-  sudo mv ~/odoo /etc/nginx/sites-available/
-  sudo ln -s /etc/nginx/sites-available/odoo /etc/nginx/sites-enabled/odoo
-  sudo rm /etc/nginx/sites-enabled/default
-  sudo service nginx reload
+  mv ~/odoo /etc/nginx/sites-available/
```
Comment (Yenthe666) [inline odoo_install.sh]: We should use `$OE_USER` for these too I believe.

File: odoo_install.sh [hunk 8]
```diff
@@ -328,10 +328,10 @@ if [ $INSTALL_NGINX = "True" ]; then
...
+  ln -s /etc/nginx/sites-available/odoo /etc/nginx/sites-enabled/odoo
```
Comment (Yenthe666) [inline odoo_install.sh]: We should use `$OE_USER` for these too I believe.

**Review by Yenthe666 — CHANGES_REQUESTED** [review]

**Comment by Yenthe666** [comment]
@tonybenoy I think we need those too. It looks neat :)

File: odoo_install.sh [hunk 9]
```diff
@@ -328,8 +328,8 @@ if [ $INSTALL_NGINX = "True" ]; then
    }
 EOF
 
-  mv ~/odoo /etc/nginx/sites-available/
-  ln -s /etc/nginx/sites-available/odoo /etc/nginx/sites-enabled/odoo
+  mv ~/$OE_USER /etc/nginx/sites-available/
+  ln -s /etc/nginx/sites-available/odoo /etc/nginx/sites-enabled/$OE_USER
```
Comment (Yenthe666) [inline odoo_install.sh:332]: Shouldn't you change the first part too? It now has /odoo hardcoded.

**Review by Yenthe666 — CHANGES_REQUESTED** [review]

**Comment by Yenthe666** [comment]
@tonybenoy can you solve that final one & rebase? I made some commits with other improvements and fixes that give issues now.
Can you give it one more test run locally maybe?

**Comment by tonybenoy** [comment]
Hey @Yenthe666 . Sorry for not responding. I was not available for the time. I'm trying to make some time to make all the changes that you suggested. I did try with the minor name changes and got some issues. So I'll do some more proper testing and get back to you :)

**Comment by Yenthe666** [comment]
Let us close this one off in favour of https://github.com/Yenthe666/InstallScript/pull/137

**Pull request closed without merging**